org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/InvisibleProjectPreferenceChangeListener.java

```diff
@@ -55,6 +63,34 @@ public void preferencesChange(Preferences oldPreferences, Preferences newPrefere
 					IPath outputPath = InvisibleProjectImporter.getOutputPath(javaProject, newPreferences.getInvisibleProjectOutputPath(), true /*isUpdate*/);
 					IClasspathEntry[] classpathEntries = InvisibleProjectImporter.resolveClassPathEntries(javaProject, sourcePaths, excludingPaths, outputPath);
 					javaProject.setRawClasspath(classpathEntries, outputPath, new NullProgressMonitor());
+					if (JavaLanguageServerPlugin.getInstance() != null && JavaLanguageServerPlugin.getInstance().getProtocol() != null && JavaLanguageServerPlugin.getInstance().getProtocol().getClientConnection() != null) {
```

Comment:
Is the JavaLanguageServerPlugin.getInstance() null check needed ? I don't think we do such a check anywhere else in the code, but the others seem fine.

Reply:
Fixed.